### ISSUE TITLE
Cannot upload multiple images when using multiple uploaders

### ISSUE THREAD

**Issue by vineethvijayan314** [FlutterFlow/flutterflow-issues#1497]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Hi team,

I am trying to upload multiple images and FF doesn't allow me to.

The scenario here is I have 2 image uploaders and I save the images locally until the user hits the save button.

We don't want unsaved images in the storage.

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/ad5acc1d-d8d6-4f37-b5ae-d2f6f99ce8b2)

### Expected Behavior

Should allow multiple image uploaders to work together.

### Steps to Reproduce

- Create a page
- Carate 2 image uploaders both image uploaders should be locally saved.
- add a save button 
- in the action of the button upload both images to supabase storage.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4ei8mBw4ZOxe5E0b6JbcEwhiAXNl87UIIvlu5+FQwcJZ+tEYl3fejxP2FoUtfiTAk7L1n8oD00psKNvd7uA+UnMk2Cfak5zLhcTTvJQkSWaKqTO82CWXMlM+ZJJVyn0cO7uBRCWbdYLFo921yEO/CRcC7dQ5P1NTMzHvL9WZjL+murQ0SLb3kNh1JWeC/v

### Context

I am trying to upload multiple images and FF doesn't allow me to.

### Visual documentation

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/8686f581-ad5b-4a0f-8c18-e0256d3f3dd8)

### Environment

```markdown
- FlutterFlow version: 4.0
- Platform: Web
- Browser name and version: Brave 
- Operating system and version affected: Windows 11
```

**Comment by hariprasadms** [comment]
Hi @vineethvijayan314 - The issue title and description is not matching the what you are expecting in the issue. Please update the issue details accordingly. 

At the moment storing the images locally before upload to backend is not by default available. However you can use custom code to achieve the same.

**Comment by vineethvijayan314** [comment]
@hariprasadms I understand it is directly not available, I have managed to do it with a single upload. I also tried putting it in a loop and based on a condition it should select the right uploaded file.

But then I get a different issue. it shows no error in FF, but when I try to run it it shows

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/fc924cd6-39db-41ad-a75d-61a473f58fb6)

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/57871dd4-8c9f-4a0b-b984-ea8df756cfe2)

**Comment by vineethvijayan314** [comment]
I have debugged more and it seems only when I give the if condition it fails to run. If in the loop I give the same uploaded file it uploads multiple times the same file. which makes sense. But I don't understand why it's failing when I give an if condition and select between multiple uploaded files.

**Comment by hariprasadms** [comment]
Hi, As per your original issue description trying to use multiple uploader in the same action flow is not possible. This is as per current design. 

<img width="878" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/15c25a60-95a3-4adb-93af-f8015956df39">

The latest issue you are facing is something different one compare to original one. I would suggest close the this issue and create new one with reproducible steps with a blank project setup with issue step details. I hope this can help you. Thakns.

**Comment by vineethvijayan314** [comment]
@hariprasadms I managed to do this with the exiting FF,

I created a temp variable with the type Uploaded File

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/90d12e4b-70cc-4d75-a3b3-960fef6e6717)

on the button save add a loop with counter < than 2 (since I have 2 uploads) then based on the loop counter index and set the tempImage as the uploaded image 1 or 2. Upload the image and get the uploaded URL. 

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/81630465/a2f1d1ee-ed14-4e8a-a697-b3f25931fc20)

I have a few more logic add like if there is a change then only upload. 

Adding this here so that anyone else looking can find the answer. It was a fun experiment. And a brain-teasing one. 

Thanks @hariprasadms
